src/transport/mod.rs

```diff
@@ -0,0 +1,101 @@
+//! The module containing the [`Transport`] struct and
+//! its [`Builder`](TransportBuilder).
+
+mod builder;
+mod http_low_level;
+
+pub use builder::TransportBuilder;
+pub use http_low_level::HttpLowLevel;
+
+/// The [`Transport`] struct.
+///
+/// It handles the low level HTTP client.
+pub struct Transport<C> {
+    client: C,
+}
+
+#[cfg(feature = "tokio_rt")]
+impl Default for Transport<http_low_level::HyperHttpsClient> {
+    fn default() -> Self {
+        TransportBuilder::new_hyper().build()
+    }
+}
+
+impl<C> Transport<C>
+where
+    C: HttpLowLevel,
+{
+    /// Make a request that will be accepted by
+    /// [`send`](Self::send) function.
+    pub fn make_request(
+        &self,
+        method: C::Method,
+        uri: &str,
+        headers: C::HeaderMap,
+        body: C::Body,
+    ) -> C::Request {
+        C::make_request(method, uri, headers, body)
+    }
+
+    /// Send a request and receive a response.
+    pub async fn send(&self, request: C::Request) -> C::Response {
+        self.client.send(request).await
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use http::Method as HttpMethod;
+    use http::{HeaderMap, StatusCode};
+    use httpmock::Method as MockMethod;
+    use httpmock::MockServer;
+
+    use super::*;
+
```

Comment:
About that, correct me if I'm wrong but I don't think MockServer would work with wasm bindgen test.

Reply:
Good point! I don't know haha. We could disable that test for that feature and look for other ways to test that functionality if that is the case.

Reply:
What if try using [`httpbin.org`](https://httpbin.org/), for the testing purposes? I don't think it would be that flexible though. Or else we could use other docker containers for http [mocking](https://hub.docker.com/r/alexliesenfeld/httpmock).

Reply:
The problem with that will be that if for some reason when we are doing the test that page has a downtime or something like that the test will fail. If we run into problems are unable to add tests for the wasm part we could  use that httpbin.org, that is the thing that is being used in https://github.com/http-rs/surf/blob/5462b3f4d47a6419338caa2ec98e4376a85e6eae/wasm-test/tests/wasm.rs#L7 but I am not too happy about that 😂

Reply:
I just readed about https://docs.rs/httpmock/0.5.8/httpmock/#standalone-mode and that looks fine, we can run that docker container in the GA

Reply:
Quick update, I don't think the tests are working in node, but they are working in the browser, so I would be updating the ci.yml too